internal/pkg/gateway/registry.go

```diff
@@ -465,3 +467,17 @@ func (reg *registry) closeStaleOrdererConnections(channel string, channelOrderer
 		}
 	}
 }
+
+func (reg *registry) hasChaincode(member gossipdiscovery.NetworkMember, chaincodeName string) bool {
+	if reg.systemChaincodes.IsSysCC(chaincodeName) {
+		return true
+	}
+
```

Comment:
Probably better to reverse the order. Application chaincode calls are much more frequent than SCC ones

Reply:
I do not think we have an alternative for [CSCC](https://github.com/hyperledger/fabric/blob/main/core/scc/cscc/configure.go#L73) to join peer channel, etc...